src/rewrite.ml

```diff
@@ -0,0 +1,235 @@
+(** Implementation of the REWRITE tactic. *)
+
+open Timed
+open Terms
+open Print
+open Console
+open Proofs
+
+(****************************************************************************)
+(* Error function. *)
+let fail_to_match = fun n ->
```

Comment:
I would avoid using this function as it is not useful, and it is really easy mix up messages. You should inline the calls to `fatal_no_pos`.